src/net/memory.c

```diff
@@ -49,6 +49,24 @@ void _z_sample_free(_z_sample_t **sample) {
     }
 }
 
+void _z_sample_copy(_z_sample_t *dst, const _z_sample_t *src) {
+    dst->keyexpr = _z_keyexpr_duplicate(src->keyexpr);
+    dst->payload = _z_bytes_duplicate(&src->payload);
+    dst->timestamp = _z_timestamp_duplicate(&src->timestamp);
+    // TODO(sashacmc): should be implemented after encoding rework
```

Comment:
So `dst->encoding` is left unitialized?

Reply:
Yes, this is not very good, I agree, I added a copy of the current encoding implementation